### PR TITLE
Check whether kited supports buffer's language

### PR DESCRIPTION
**Pull request by airblade**
See #250.

### CODE REVIEW

**Comment by ErrEoE** [comment]
If we provide settings for supported languages,  such as `let g:kite_support_language=['python','go']`, it may be more efficient and flexible.

**Comment by airblade** [comment]
@jansorg and I discussed this and he wanted to avoid introducing another variable for testers to keep track of when activating/deactivating Go.

**Comment by ErrEoE** [comment]
We get language support from server on `BufEnter`,it may be slowing buffer switch.

**Comment by airblade** [comment]
@ErrEoE 

> We get language support from server on `BufEnter`

Only once.

**Comment by jansorg** [comment]
There's be a flag in the kite plugin to control if Go is enabled or not

**Comment by ErrEoE** [comment]
Thank you for your reply, there is no any problem.

**Comment by airblade** [comment]
@jansorg It's ready for another review now.

File: autoload/kite/languages.vim [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  return 0
+endfunction
+
+
+" Returns true if the current buffer's language is supported by kited, false otherwise.
+function! kite#languages#supported_by_kited()
+  " Only check kited's languages once.
+  if empty(s:languages_supported_by_kited)
+    let s:languages_supported_by_kited = kite#client#languages(function('kite#languages#handler'))
```
Comment (jansorg) [inline autoload/kite/languages.vim]: Which set of languages is assumed when this requests fails?

File: autoload/kite.vim [hunk 2]
```diff
@@ -87,24 +87,28 @@ endfunction
...
 function! kite#bufenter()
-  if s:supported_language()
+  if kite#languages#supported_by_plugin()
     call s:launch_kited()
 
-    call s:disable_completion_plugins()
-    call s:setup_options()
-    call s:setup_events()
-    call s:setup_mappings()
+    if kite#languages#supported_by_kited()
```
Comment (jansorg) [inline autoload/kite.vim]: Is my understanding of the new logic correct?
- the plugin has a flag for the enabled languages, this defaults to Python=on and Go=off
- Support is only enabled if both the plugin and kited report a language as supported
- kited's language support is only requested once for the current process

Comment (airblade) [reply autoload/kite.vim]: Yes, that's all correct.

File: autoload/kite/languages.vim (hunk 1 above)
Comment (airblade) [reply autoload/kite/languages.vim]: No languages, but the code will keep trying kited (when entering a buffer) until it receives a non-empty response.

**Comment by airblade** [comment]
Rebased and merged in 52c877d...14f05e1.

**Pull request closed without merging**